src/Scheduler/SchedulerConfig.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console\Scheduler;
+
+use InvalidArgumentException;
+use ReflectionMethod;
+use Tempest\Console\ConsoleCommand;
+
+final class SchedulerConfig
+{
+    public function __construct(
+        public string $path = "php tempest",
+        public string $output = "/dev/null",
+        public OutputMode $outputMode = OutputMode::Append,
+
+        /** @var ConsoleCommand[] $schedules */
+        public array $schedules = [],
+    ) {
+    }
+
+    public function addSchedule(ReflectionMethod $handler, ConsoleCommand $command): self
+    {
+        if (! $command->schedule) {
+            throw new InvalidArgumentException('Command does not have a schedule');
+        }
+
+        $command->setHandler($handler);
```

Comment:
Why do we need to set the handler here? Isn't that done while building the console config object?

Reply:
The way it works now is that discovery of schedules is independent from the ConsoleConfig, this will be more useful if we decide to go the "attribute over any function" path

Reply:
Gotcha. Maybe then it makes more sense to keep everything within one discovery, not sure.

Let's keep it as is for now, we can always refactor later :)